Eatery/Controllers/Eateries/Campus/CampusEateriesSearchViewController.swift

```diff
@@ -6,22 +6,352 @@
+        var searchResults: [SearchResult] = []
+
+        for eatery in eateries {
+            var items = Set(eatery.diningItems(onDayOf: Date()))
+            if let activeEvent = eatery.activeEvent(atExactly: Date()) {
+                items.formUnion(activeEvent.menu.data.values.flatMap { $0 })
+            }
+
+            let filteredItems = items
+                .filter { matches($0.name, searchText) }
+                .sorted { $0.name < $1.name }
+
+            searchResults += filteredItems.map {
+                SearchResult(
+                    source: .menuItem(eatery, $0),
+                    title: $0.name,
+                    subtitle: eatery.displayName,
+                    isFavorite: false)
+            }
+        }
+
+        return searchResults
+    }
+
+    private func areaSearchResults(searchText: String) -> [SearchResult] {
+        Area.allCases
+            .filter { area in matches(area.description, searchText) }
+            .map { area in
+                SearchResult(
+                    source: .area(area),
+                    title: area.description,
+                    subtitle: "Location",
+                    isFavorite: nil)
+        }
+    }
+
+    private func matches(_ textToSearch: String, _ searchText: String) -> Bool {
+        return textToSearch.localizedCaseInsensitiveContains(searchText)
+    }
+
+}
+
 class CampusEateriesSearchViewController: UIViewController {
 
+    // Model
+
+    private var mode: Mode = .recentSearches
+    private let searchResults = SearchResultsManager()
+
+    private var displayedRecentSearches: [RecentSearch] = []
+    private var displayedSearchResults: [SearchResult] = []
+
+    // Views
+
+    private var titleLabel: UILabel!
+    private var clearButton: UIButton!
+
+    private var tableView: UITableView!
+
+    weak var searchController: UISearchController?
+
+    private var searchDebounceTimer: Timer?
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
         view.backgroundColor = .white
+
+        setUpHeaderViews()
+        setUpTableView()
+
+        setMode(.recentSearches)
+
+        displayedRecentSearches = Defaults[\.campusRecentSearches]
+        NetworkManager.shared.getCampusEateries(useCachedData: true) { (campusEateries, _) in
+            self.searchResults.eateries = campusEateries ?? []
+        }
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardFrameWillChange(_:)),
+            name: NSNotification.Name.UIKeyboardWillChangeFrame,
+            object: nil)
+    }
+
+    @objc private func keyboardFrameWillChange(_ notification: Notification) {
+        guard let userInfo = notification.userInfo,
+            let keyboardFrame = userInfo[UIKeyboardFrameEndUserInfoKey] as? CGRect else {
+                return
+        }
+
+        let keyboardFrameInView = view.convert(keyboardFrame, from: nil)
```

Comment:
I don't think you need this conversion. I think the view's frame is already equal to the window's. Although I guess maybe it's good just to be sure.